### PR TITLE
[fix] Avoid lazy parsing concurrency issues in NONMEM `Record`

### PR DESCRIPTION
**Pull request by pharmpy-dev-123**
> @rikardn Ready for merge.

### CODE REVIEW

File: src/pharmpy/plugins/nonmem/records/record.py [hunk 1]
```diff
@@ -10,18 +10,13 @@ class Record:
     def __init__(self, content, parser_class):
         self._content = content
         self._parser_class = parser_class
-        self._root = None
+        self._root = self._parser_class(self._content).root
```
Comment (pharmpy-dev-123) [inline src/pharmpy/plugins/nonmem/records/record.py:13]: Is it really necessary to keep a reference to `content` and `parser_class`? These will not be reused.

Comment (rikardn) [reply src/pharmpy/plugins/nonmem/records/record.py:13]: Not if we are parsing directly

Comment (pharmpy-dev-123) [reply src/pharmpy/plugins/nonmem/records/record.py:13]: OK removing those then.

**Review by rikardn — APPROVED** [review]
This looks good. We might need to revisit sooner than we think though. The main problem with always parsing might not the performance as I first thought but that fact that our grammars might not be able to parse all records in the wild. In some cases all records are not needed by Pharmpy but now they still need to be parsed potentially crashing the parser unnecessarily for more models.

**Pull request merged**